### ISSUE TITLE
Add support for Neo4j

### ISSUE THREAD

**Issue by Analect** [rich-iannone/DiagrammeR#94]
Just came across your tool in: http://www.r-bloggers.com/rstudio-v0-99-preview-graphviz-and-diagrammer/

I was wondering if it would be feasible to add support for visualizing the results of queries to a Neo4j database.
There is a transaction end-point in their API that returns a query in graph form, by specifying resultDataContents as "graph".  See here: 
and here:  http://neo4j.com/docs/stable/rest-api-transactional.html#rest-api-return-results-in-graph-format
http://neo4j.com/developer/guide-data-visualization/#_howto_graph_visualization_step_by_step

There is an existing RNeo4j that appears to support transactions, although I don't specifically see a worked example with results returned as a graph in the docs.
http://nicolewhite.github.io/RNeo4j/docs/transactions.html

Not much of an R programmer, but can try and help with any further clarifcations you might need.

**Comment by rich-iannone** [comment]
I'd love to get this type of feature in DiagrammeR. I'll read up on the material in those links and get back to you soon about the next step in building that in.

**Comment by abresler** [comment]
Yes!!


----------------------
*Alex Bresler*
abresler@asbcllc.com

​www.asbcllc.com​

917-455-0239
​ (cell)​

@abresler <http://twitter.com/abresler> | LinkedIn
<http://www.linkedin.com/in/abresler> | My Musings <http://asbcllc.com/blog>

**Comment by rich-iannone** [comment]
@Analect would you be able to send over two examples of Neo4j output as well as screenshots of the graph (for fairly simple graph)? The output didn't look too difficult to parse but I just want to make sure there's obvious errors when starting off.

**Comment by abresler** [comment]
Lots of examples here
http://nicolewhite.github.io/

Issue is you either need a local or remote installation of NEO4J to
interact with it, that said it'd powerful and amazing and would tie GREAT
with all the work put into diagrammer

----------------------
Alex Bresler
abresler@asbcllc.com
​www.asbcllc.com​
917-455-0239​ (cell)​

**Comment by Analect** [comment]
@rich-iannone 
Yes, Nicole White, a data scientist at Neo4j has many good examples as an intro to Neo4j.  This video walks you through her workflow with RNeo4j. https://www.youtube.com/watch?v=bdQ90y9Pefo
Getting a Neo4j instance up and running, listening on port 7474 is easy too.  Not sure if you work with Docker, but that's perhaps the quickest route to getting the latest 2.2.1 version up and running:
```
docker run -d -p 0.0.0.0:7474:7474 -v /home/path/to/data/neo4j/data:/opt/data --name neo4j kbastani/docker-neo4j
```
Then the various scripts Nicole uses in that video are contained here: https://github.com/nicolewhite/graphs_r_cool

The 'manual_create.R' will get you the graph below, when you call browse(graph).  This leverages the front-end built-in to Neo4j that leverages D3.

![image](https://cloud.githubusercontent.com/assets/4063815/7582146/bc8b7298-f882-11e4-8dbe-fcd7e208e417.png)

I mentioned your package to Nicole at GraphConnect in London last week, so perhaps reach out to her if you need some direct input.

**Comment by rich-iannone** [comment]
@Analect Sorry for the delay, but, thanks for the great info. I'm going to check out the video and run some of those examples. Really great stuff!

**Comment by scizmeli** [comment]
Any advances with this? I could not find any reference to Neo4j in the current code & documentation.
